### PR TITLE
Add build script for 3scale/zync

### PR DESCRIPTION
**Pull request by Abhijit-Mane**
PR merges 3scale Zync - The sync tool

Zync takes your 3scale data and push it somewhere else, reliably. Offers only one directional sync (from 3scale to other systems).
(https://github.com/3scale/zync )

Signed-off-by: Abhijit Mane abhijman@in.ibm.com

### CODE REVIEW

File: 3/3scale-zync/redhat_ubi8/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+MAINTAINER Abhijit Mane <abhijman@in.ibm.com>
+
+RUN yum install -y make gcc wget git gem ruby ruby-devel redhat-rpm-config \
+    shared-mime-info.ppc64le zlib.ppc64le zlib-devel.ppc64le \
+ && gem install bundler
+
+RUN dnf install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-8-ppc64le/pgdg-redhat-repo-latest.noarch.rpm \
+ && dnf install -y postgresql13-server \
+ && yum install -y libpq5 libpq5-devel
```
Comment (krishvoor) [inline 3/3scale-zync/redhat_ubi8/Dockerfile:11]: Can you keep either dnf or yum, to eliminate confusion?

File: 3/3scale-zync/redhat_ubi8/build.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# gem install mimemagic -v 0.3.10
+# bundle update mimemagic
+
+# Change "mimemagic (0.3.5)" to "mimemagic (0.3.10) in zync/Gemfile.lock
+############################
+
+
+# install & setup
+bundle install
+./bin/setup
```
Comment (krishvoor) [inline 3/3scale-zync/redhat_ubi8/build.sh:59]: Do we plan to copy out the resultant build artefact out to /ws/?

**Review by krishvoor — COMMENTED** [review]
Minor queries, otherwise /lgtm

File: 3/3scale-zync/redhat_ubi8/Dockerfile (hunk 1 above)
Comment (gerrith3) [reply 3/3scale-zync/redhat_ubi8/Dockerfile:11]: Stick to dnf, please.

File: 3/3scale-zync/redhat_ubi8/build.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# ----------------------------------------------------------------------------
+# Package       : zync
+# Version       : master
```
[the remark this answers is not in the document]
Comment (Abhijit-Mane) [reply 3/3scale-zync/redhat_ubi8/build.sh:3]: There is some explanation in the build script where 3scale-2.9.1-GA and 3scale-2.10.0-GA releases are locked to mimemagic (0.3.5) in zync/Gemfile.lock. This version is no longer supported and bundle install fails.
mimemagic (0.3.10) needs to be used and the master has these changes. The upcoming release should incorporate this.
Hence, currently build is off master.

######## needed only for below releases:
#BRANCH=3scale-2.10.0-GA
#BRANCH=3scale-2.9.1-GA

# This update is needed before install
# gem install mimemagic -v 0.3.10
# bundle update mimemagic

# Change "mimemagic (0.3.5)" to "mimemagic (0.3.10) in zync/Gemfile.lock
############################

File: 3/3scale-zync/redhat_ubi8/Dockerfile (hunk 1 above)
Comment (Abhijit-Mane) [reply 3/3scale-zync/redhat_ubi8/Dockerfile:11]: Done

File: 3/3scale-zync/redhat_ubi8/build.sh (hunk 2 above)
Comment (Abhijit-Mane) [reply 3/3scale-zync/redhat_ubi8/build.sh:59]: Done

File: 3/3scale-zync/redhat_ubi8/build.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+then
+	bundle exec rails test
+fi
+
+# copy
+cp -r vendor/bundle ../
+
+# cleanup
+cd ..
+#rm -rf zync
```
Comment (krishvoor) [inline 3/3scale-zync/redhat_ubi8/build.sh:72]: We are not "removing" as it is not part of `/ws/` ?

**Review by krishvoor — COMMENTED** [review]
We not cleaning up the `workspace` as it part of `/ws`?
Otherwise, /lgtm

**Review by afsanjar — APPROVED** [review]

File: 3/3scale-zync/redhat_ubi8/build.sh (hunk 4 above)
Comment (Abhijit-Mane) [reply 3/3scale-zync/redhat_ubi8/build.sh]: Good catch. I commented for testing so the 'zync' directory wouldn't disappear before I get a chance to check its contents :)
Had missed to uncomment before commit. Fixed now.

**Pull request merged**